pybossa/cache/users.py

```diff
@@ -357,7 +357,6 @@ def get_user_preferences(user_id):
     return get_user_pref_db_clause(user_pref, user_email)
 
 
-@memoize(timeout=ONE_DAY)
```

Comment:
Instead of removing memoize, is it possible to call delete_memoize on get_user_filters whenever db update happens for this user_id. Similar implementation for n_tasks can be referred [here](https://github.com/bloomberg/pybossa/blob/master/pybossa/cache/projects.py#L152-L153) and [here](https://github.com/bloomberg/pybossa/blob/master/pybossa/cache/projects.py#L644-L646)

Reply:
This is what I thought at the beginning, but now I feel like removing cache is not a big deal as it doesn't add traffic to database

Reply:
IMO, whatever fix we make for `get_user_filter` should also be added to [this function].(https://github.com/bloomberg/pybossa/blob/c09b982006966c4a4ba0b6ab996f11f875fa63e7/pybossa/cache/users.py#L353), both of them are used in user_preference scheduler

Reply:
As I explained in the above comment, I think removing the cache is the correct way, because "get_user_filters" is a wrapper function of "get_user_by_id" and "get_user_by_id" is already cached.

Reply:
Accessing data from cache is always faster than db irrespective of amount of traffic it adds to db or not. Hence, if its not too much effort, I would vote for resetting cache whenever db update happens to the attributes.

Reply:
> IMO, whatever fix we make for `get_user_filter` should also be added to [this function](https://github.com/bloomberg/pybossa/blob/c09b982006966c4a4ba0b6ab996f11f875fa63e7/pybossa/cache/users.py#L353), both of them are used in user_preference scheduler

"get_user_preferences"  also has duplicate cache, it calls "get_user_by_id" and "get_user_by_id" is already cached. The reason it works is because "[delete_user_pref_metadata](https://github.com/bloomberg/pybossa/blob/3da998fcb56489bb3ba9c9e0f42b4fa2faeb10e5/pybossa/cache/users.py#L346)" clears it.

Reply:
"get_user_preferences"  also caches old data query for 24 hours like "get_user_filters" does, I think we should make them consistent by removing `@memoize(timeout=ONE_DAY)` since both are using for user preference scheduler, what do you think?

Reply:
Although they way "get_user_preferences" did can work, I agree to make it correct as well as consistent.